### PR TITLE
fix: remove deprecated removeListener methods

### PR DESCRIPTION
**Pull request by matinzd**
## Summary
Remove old deprecated modules that cause annoying warnings. This can be a breaking change for some third-party modules. 

## Changelog

[General] [Removed] - Remove deprecated removeListener methods

### CODE REVIEW

File: packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js [hunk 1]
```diff
@@ -1164,12 +1164,15 @@ class DisplayOptionsStatusExample extends React.Component<{}> {
 function DisplayOptionStatusExample({optionName, optionChecker, notification}) {
   const [statusEnabled, setStatusEnabled] = React.useState(false);
   React.useEffect(() => {
-    AccessibilityInfo.addEventListener(notification, setStatusEnabled);
+    let listener = AccessibilityInfo.addEventListener(
```
Comment (thomaswinckell) [inline packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1167]: It's better to you `const` instead of `let` here

Comment (matinzd) [reply packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js:1167]: I could see some issues in linting while using const in different scopes. I will change it to const.

Comment (matinzd) [reply packages/rn-tester/js/examples/Accessibility/AccessibilityExample.js]: Fixed [ad7a226](https://github.com/facebook/react-native/pull/33580/commits/ad7a226bdacecea557001876dc7dcbcfb82075a1).

**Pull request closed without merging**